nucleus/common/common-util/src/main/java/com/sun/enterprise/loader/ASURLClassLoader.java

```diff
@@ -955,88 +944,21 @@ void init() throws IOException {
                 }
 
-                table = new ConcurrentHashMap<>();
             } catch (URISyntaxException use) {
                 throw new IOException(use);
             }
         }
 
-        private void fillTable(File f, Map<String, String> t, String parent) throws IOException {
-            String localName = (parent.equals("")) ? "" : parent + "/";
-            File[] children = f.listFiles();
-            if (children != null) {
-                for (File aChildren : children) {
-                    processFile(aChildren, t, localName);
-                }
-            }
-        }
-
-        /**
-         *Adds a file (or, if a directory, the directory's contents) to the table
-         *of files this loader knows about.
-         *<p>
-         *Invokes fillTable for subdirectories which in turn invokes processFile
-         *recursively.
-         *@param fileToProcess the File to be processed
-         *@param t the Map that holds the files the loader knows about
-         *@param parentLocalName prefix to be used for the full path; should be
-         *non-empty only for recursive invocations
-         *@throws IOException in case of errors working with the fileToProcess
-         */
-        private void processFile(File fileToProcess, Map<String, String> t, String parentLocalName) throws IOException {
-            String key = parentLocalName + fileToProcess.getName();
-            if (fileToProcess.isFile()) {
-                t.put(key, key);
-            } else if (fileToProcess.isDirectory()) {
-                fillTable(fileToProcess, t, key);
-            }
-        }
-
-
         boolean hasItem(String item) {
-            // in the case of ejbc stub compilation, asurlclassloader is created before stubs
-            // gets generated, thus we need to return true for this case.
-            if (table.isEmpty()) {
-                return true;
-            }
-
-            /*
-             *Even with the previous special handling, a file could be created
-             *in a directory after the loader was created and its table of
-             *URLEntry names populated.  So check the table first and, if
-             *the target item is not there and this URLEntry is for a directory, look for
-             *the file.  If the file is now present but was not when the loader
-             *was created, add an entry for the file in the table.
-             */
-            String target = item;
-            // special handling
-            if (item.startsWith("./")) {
-                target = item.substring(2);
-            }
-
-            boolean result = table.containsKey(target);
-            if ( ! result && ! isJar) {
-                /*
-                 *If the file exists now then it has been added to the directory since the
-                 *loader was created.  Add it to the table of files we
-                 *know about.
-                 */
-                File targetFile = privilegedCheckForFile(target);
```

Comment:
It's hard to tell because this codepath was not invoked for at least 8 years. But I can add privileged actions around filesystem access

Reply:
I am happy to gamble on something that wasn't used in 8 years and not do any more than you already did. But that is just me. Don't know @Pandrex247 thinks about it.

Reply:
I would like to see all TCK tests results on this, because classloading magic changed over time ... in SGES was strict classpath prefix and suffix, isolated untouchable endorsed, and fragile JNDI/classloading issues with EAR/WAR/EJB(outer/inner) scopes. Then came GF3 and everything changed. Javadoc of this class from 2011 says `@since  JDK 1.4`, so I would expect it was something around SGES and perhaps even SJAS7 mechanisms.

Simply said: it is a zombie :D

Btw on line 826 of original file is incorrect string concatenation and also append("\n") is slower than append('\n'). And hasItem manipulates it's parameter - it is used in cycle so it does it again and again until JIT maybe would optimize it. Blah.